tests/integration/components/selection-test.js

```diff
@@ -536,6 +537,67 @@ module('Integration | selection', () => {
           'After scrolling to bottom, there are still no rows selected'
         );
       });
+
+      test('Issue 747: Programmatically select an un-rendered row', async function(assert) {
+        let rows = generateRows(200, 1);
+        await generateTable(this, { rows, bufferSize: 1 });
+
+        let renderedRowCount = table.rows.length;
+        assert.ok(renderedRowCount < 200, 'some rows are occluded');
+
+        // Select rows at the end that will not all have been rendered yet
+        this.set('selection', rows.slice(-5));
+
+        await table.rows.objectAt(0).checkbox.click();
+        assert.ok(true, 'no error');
```

Comment:
Should this also assert that the row is selected when you scroll to it?

Reply:
I'll add an assertion that it is selected, but this row is at the top of the table so will already be in view.